src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackEncoder.cs

```diff
@@ -217,16 +217,28 @@ private static bool EncodeStringLiteralValue(string value, Span<byte> destinatio
         {
             if (value.Length <= destination.Length)
             {
-                for (int i = 0; i < value.Length; i++)
+                if (HttpConnectionSettings.AllowNonAsciiHeaders)
                 {
-                    char c = value[i];
-                    if ((c & 0xFF80) != 0)
```

Comment:
You should still filter by `0xFF00`. Downcasting a character in that range to a byte will cause data corruption. Only the 0xFF range can be cast to a byte.

Reply:
Is this addressed?

Reply:
It was at some point but it seems to have regressed again.

Reply:
Nevermind, this links to old code.